bazel/toolchain/BUILD.bazel

```diff
@@ -168,6 +168,7 @@ cc_toolchain_config(
     # See https://github.com/bazelbuild/bazel/issues/15875.
     ar_flags = "rcs",
     cc = "/usr/bin/clang",
+    cc_flags = ["-flto=thin"],
```

Comment:
won't this also apply to debug builds. "--config=dbg"?

Reply:
I'm getting tired of the .bazelrc stuff for switching between debug/release builds. Would probably rather define two separate targets: :urbit and :urbit-dbg. Would then include -flto=thin in the :urbit target

Reply:
Agreed. My experience trying to make Bazel do anything that I want it to do has been nightmarish. I encourage using whatever grug-brain approach will get us easily-configurable debug and release builds quickly, instead of trying to navigate Google's galaxy brain dogshit docs.

Reply:
That said, I think this PR should just be merged as-is, with a an issue opened to add LTO to the Linux build, and possibly another issue to separate the dbg build into a dbg target.

Reply:
Only thing preventing merge right now is -flto needs to be constrained to release builds. I don't want this optimization on -O0 debug builds. And agree we can merge the macos changes before linux. We ran into some issue with libsigsegv when attempting -flto on all dependencies. Best we've got right now on linux is to constrain it to just vere compilation units

Reply:
Yeah, constraining to release is coming, I'll push tomorrow if plans go according to plan :p